web/src/topic_list.ts

```diff
@@ -181,12 +193,32 @@ export function is_complete_for_stream_id(stream_id: number): boolean {
             return true;
         }
 
-        // Now, we can just compare the first cached message to the first
-        // message ID in the stream; if it's older, we're good, otherwise,
-        // we might be missing the oldest topics in this stream in our
-        // cache.
         const first_cached_message = all_messages_data.first_including_muted();
-        return first_cached_message!.id <= sub.first_message_id;
+        if (sub.first_message_id < first_cached_message!.id) {
+            // Missing the oldest topics in this stream in our cache.
+            return false;
+        }
```

Comment:
I think this check might be inverted? The thing that is true is that if `sub.first_message_id === first_cached_message!.id`, then we can return `true`, because we've seen we have complete history.

But if we shouldn't return `false` here -- if we do, I don't think we'll ever reach the code below.